freud/locality.pyx

```diff
@@ -524,6 +524,88 @@ cdef class NeighborList:
 
         return result
 
+    @classmethod
+    def from_points(
+            cls, box, points, query_points, query_point_indices,
+            point_indices, weights=None
+    ):
+        R"""Create a NeighborList from a set of bond information arrays
+        when distances are not known.
+
+        Args:
+            box (:class:`~.box.Box`):
+                Simulation box.
+            points (:class:`np.ndarray`):
+                Array of floats corresponding to the positions of a set of points.
+            query_points (:class:`np.ndarray`):
+                Array of floats corresponding to the positions of a set of query points.
+            query_point_indices (:class:`np.ndarray`):
+                Array of integers corresponding to indices in the set of
+                query points.
+            point_indices (:class:`np.ndarray`):
+                Array of integers corresponding to indices in the set of
+                points.
+            weights (:class:`np.ndarray`, optional):
+                Array of per-bond weights (if :code:`None` is given, use a
+                value of 1 for each weight) (Default value = :code:`None`).
+        """  # noqa 501
+
+        query_point_indices = np.asarray(query_point_indices)
+        point_indices = np.asarray(point_indices)
+
+        if query_point_indices.shape != point_indices.shape:
+            raise ValueError(
+                "Points and query points indices have shapes {} and {}".format(
```

Comment:
You probably need to combine these strings -- it looks like you're passing the message as two separate arguments to `ValueError`.
```suggestion
                "Points and query points indices have shapes {} and {}. ".format(
```